src/System.Private.CoreLib/shared/System/LocalAppContextSwitches.Common.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static partial class LocalAppContextSwitches
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool GetCachedSwitchValue(string switchName, ref int switchValue)
```

Comment:
I assume these are ints only because we need the "not-cached" value. If perf doesn't matter much here, would it be cleaner to store these as Nullable<bool> ?

Reply:
But it's not important - if we keep the ints, please add a comment describing what the allowed values are (and what they mean).

Reply:
> would it be cleaner to store these as Nullable

Nullable access is not atomic. We would need to take lock or something to access it. A more self-describing option would be enum.

I assume that these were ints to make this code lean and simple. It is duplicated in each assembly that has these switches. (I have not written this - this was in AppContextSwitches.cs originally.)

I will add comment.

Reply:
Good point about the atomic access. I was commenting on the code since this is the first time I see it... and it took me a while to figure out why ints and not just bools.